Cython/Compiler/Parsing.py

```diff
@@ -131,49 +131,64 @@ def p_lambdef(s, allow_conditional=True):
 #lambdef_nocond: 'lambda' [varargslist] ':' test_nocond
 
 def p_lambdef_nocond(s):
-    return p_lambdef(s, allow_conditional=False)
+    return p_lambdef(s)
 
 #test: or_test ['if' or_test 'else' test] | lambdef
 
-def p_test(s, allow_assignment_expression=True):
+def p_test(s):
+    # The check for a following ':=' is only for error reporting purposes.
+    # It simply changes a
+    #   expected ')', found ':='
+    # message into something a bit more descriptive.
+    # It is close to what the PEG parser does in CPython, where an expression has
+    # a lookahead assertion that it isn't followed by ':='
+    expr = p_test_no_namedexpr_check(s)
+    if s.sy == ':=':
+        s.error("invalid syntax: assignment expression not allowed in this context")
+    return expr
+
+def p_test_no_namedexpr_check(s):
```

Comment:
Why the word `check` here? Either a named expression is allowed or not, but if it's only being checked for, what does that mean?

Reply:
I think I meant "p_test without the `if.sy == ':=': s.error("Invalid syntax...")` check".

I see your point about the name - it isn't the best

Reply:
Right. I've renamed to `p_test_allow_walrus_after`. I'm still not convinced that's the best name ever but I think it reads a bit more clearly and I'm genuinely struggling to think of anything better.